apis/pipelines/v1alpha4/depending_on_pipeline_resource.go

```diff
@@ -0,0 +1,8 @@
+package v1alpha4
+
+// +kubebuilder:object:generate=false
+type DependingOnPipelineResource interface {
```

Comment:
This is needed to allow handling dependencies from Runs and RunConfigurations to Pipelines in a generic way

Reply:
In the future, this could become generic on the dependency type